app/scripts/migrations/078.test.js

```diff
@@ -0,0 +1,460 @@
+            },
+            'uuid-4': {
+              chainId: '0x38',
+              chainName:
+                'BNB Smart Chain (previously Binance Smart Chain Mainnet)',
+              rpcPrefs: {
+                blockExplorerUrl: 'https://bscscan.com/',
+              },
+              rpcUrl: 'https://bsc-dataseed.binance.org/',
+              ticker: 'BNB',
+            },
+          },
+        },
+      },
+    };
+    const newStorage = await migration78.migrate(oldStorage);
+    expect(newStorage).toStrictEqual({
+      meta: {
+        version: 78,
+      },
+      data: {
+        PreferencesController: {
+          transactionSecurityCheckEnabled: false,
+          useBlockie: false,
+          useCurrencyRateCheck: true,
+          useMultiAccountBalanceChecker: true,
+          useNftDetection: false,
+          useNonceField: false,
+        },
+        NetworkController: {
+          network: '1',
+          networkDetails: {
+            EIPS: {
+              1559: true,
+            },
+          },
+          previousProviderStore: {
+            chainId: '0x89',
+            nickname: 'Polygon Mainnet',
+            rpcPrefs: {},
+            rpcUrl:
+              'https://polygon-mainnet.infura.io/v3/373266a93aab4acda48f89d4fe77c748',
+            ticker: 'MATIC',
+            type: 'rpc',
+          },
+          provider: {
+            chainId: '0x1',
+            nickname: '',
+            rpcPrefs: {},
+            rpcUrl: '',
+            ticker: 'ETH',
+            type: 'mainnet',
+          },
+          networkConfigurations: {
+            'uuid-1': {
+              chainId: '0x539',
+              chainName: 'Localhost 8545',
+              rpcPrefs: {},
+              rpcUrl: 'http://localhost:8545',
+              ticker: 'ETH',
+            },
+            'uuid-2': {
+              chainId: '0xa4b1',
+              chainName: 'Arbitrum One',
+              rpcPrefs: {
+                blockExplorerUrl: 'https://explorer.arbitrum.io',
+              },
+              rpcUrl:
+                'https://arbitrum-mainnet.infura.io/v3/373266a93aab4acda48f89d4fe77c748',
+              ticker: 'ETH',
+            },
+            'uuid-3': {
+              chainId: '0x4e454152',
+              chainName: 'Aurora Mainnet',
+              rpcPrefs: {
+                blockExplorerUrl: 'https://aurorascan.dev/',
+              },
+              rpcUrl:
+                'https://aurora-mainnet.infura.io/v3/373266a93aab4acda48f89d4fe77c748',
+              ticker: 'Aurora ETH',
+            },
+            'uuid-4': {
+              chainId: '0x38',
+              chainName:
+                'BNB Smart Chain (previously Binance Smart Chain Mainnet)',
+              rpcPrefs: {
+                blockExplorerUrl: 'https://bscscan.com/',
+              },
+              rpcUrl: 'https://bsc-dataseed.binance.org/',
+              ticker: 'BNB',
+            },
+          },
+        },
+      },
+    });
```

Comment:
Since you check for it in the migration, is it worth adding a test for when there is no PreferencesController state whatsoever?

Reply:
added [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/14e17a724380a89fc3155c6a45dc5ee9a5847f7a)